pkg/controller/infrastructure/infraflow/ensure.go

```diff
@@ -612,3 +632,34 @@ func (fctx *FlowContext) ensureKubernetesRoutesDeleted(ctx context.Context) erro
 
 	return nil
 }
+
+func (fctx *FlowContext) ensureAliasIpRanges(ctx context.Context) error {
+	log := shared.LogFromContext(ctx)
+
+	for i, route := range fctx.state.Routes {
+		// if machine is already gone procceed with next route
+		if err := fctx.computeClient.InsertAliasIPRoute(ctx, route, DefaultSecondarySubnetName); err != nil && !strings.Contains(err.Error(), "notFound") {
+			log.Info("Failed to add alias IP", "route", route.DestinationCIDR, "instance", route.InstanceName, "error", err)
+			perr := fctx.persistState(ctx)
+			if perr != nil {
+				fctx.log.Error(perr, "failed to persist state")
+			}
+			return err
+		} else {
+			log.Info("Successfully added alias IP", "route", route.DestinationCIDR, "instance", route.InstanceName)
+			if i >= 0 && i < len(fctx.state.Routes) {
+				if i < len(fctx.state.Routes)-1 {
+					fctx.state.Routes = append(fctx.state.Routes[:i], fctx.state.Routes[i+1:]...)
```

Comment:
I will be honest that I need to touch up on the go semantics here since we are not only modifying the slice contents but the actual slice header by changing it's length. If you know about it I would be curious of the effects albeit I would prefer avoiding altogether.